sqlmesh/core/renderer.py

```diff
@@ -325,10 +324,21 @@ def time_column_filter(self, start: TimeLike, end: TimeLike) -> exp.Between:
 
     def _optimize_query(self, query: exp.Subqueryable) -> exp.Subqueryable:
         # We don't want to normalize names in the schema because that's handled by the optimizer
-        schema = ensure_schema(self.schema, dialect=self._dialect, normalize=False)
+        schema = MappingSchema(self.schema, dialect=self._dialect, normalize=False)
         original = query
         failure = False
 
+        if not schema.empty:
+            for dependency in d.find_tables(query, dialect=self._dialect):
+                if schema.find(exp.to_table(dependency, dialect=self._dialect)) is None:
+                    logger.warning(
+                        "Query cannot be optimized due to missing schema for model '%s'. "
```

Comment:
Can we explain more what we mean by "cannot be optimized"? I could see someone reading this and thinking "Oh that is ok my engine does optimizations anyways" and missing the real negative of this. Also I wonder if having a dedicated doc talking about this could be useful (but probably outside the scope of this PR).

Reply:
I wonder what the real negative of this at runtime anyway?

Reply:
At compile time we catch this earlier when we set the mapping schema as part of the `_Model.update_schema`.